.github/CONTRIBUTING.md

```diff
@@ -1,41 +1,68 @@
 
-* You need a [GitHub account](https://github.com/signup/free)
-* Submit an [issue ticket](https://github.com/hashicorp/vagrant/issues) for your issue if there is not one yet.
-	* Describe the issue and include steps to reproduce when it's a bug.
-	* Ensure to mention the earliest version that you know is affected.
-  * If you plan on submitting a bug report, please submit debug-level logs along
-    with the report using [gist](https://gist.github.com/) or some other paste
-    service by prepending `VAGRANT_LOG=debug` to your `vagrant` commands.
-* Fork the repository on GitHub
+Before opening a new issue or pull request, we do appreciate if you take some time to search for [possible duplicates](https://github.com/hashicorp/vagrant/issues?q=sort%3Aupdated-desc), or similar discussions in the [mailing list](https://groups.google.com/forum/#!forum/vagrant-up). On GitHub, you can scope searches by labels to narrow things down.
 
-## Make Changes
+## Issues
+
+### Reporting an Issue
+
+**Tip:** We have provided a [GitHub issue template](https://github.com/hashicorp/vagrant/blob/master/.github/ISSUE_TEMPLATE.md). By respecting the proposed format and filling all the relevant sections, you'll strongly help the Vagrant collaborators to handle your request the best possible way.
+
+### Issue Lifecycle
+
+1. The issue is reported.
+2. The issue is verified and categorized by Vagrant collaborator(s). Categorization is done via GitHub tags for different dimensions (like issue type, affected components, pending actions, etc.)
+3. Unless it is critical, the issue is left for a period of time, giving outside contributors a chance to address the issue. Later, the issue may be assigned to a Vagrant collaborator and planned for a specific release [milestone](https://github.com/hashicorp/vagrant/milestones)
+4. The issue is addressed in a pull request or commit. The issue will be referenced in the commit message so that the code that fixes it is clearly linked.
+5. The issue is closed. Sometimes, valid issues will be closed to keep the issue tracker clean. The issue is still indexed and available for future viewers, or can be re-opened if necessary.
+
+## Pull Requests
+
+Thank you for contributing! Here you'll find information on what to include in your Pull Request (“PR” for short) to ensure it is reviewed quickly, and possibly accepted.
+
+Before starting work on a new feature or anything besides a minor bug fix, it is highly recommended to first initiate a discussion with the Vagrant community (either via a GitHub issue, the [mailing list], IRC freenode `#vagrant` or [Gitter](https://gitter.im/mitchellh/vagrant)). This will save you from wasting days implementing a feature that could be rejected in the end.
```

Comment:
Is the `[mailing list]` block here missing the URL?

Reply:
@briancain Thanks for the good catch! 

:bookmark_tabs: I actually noticed that I was about to forget the same URL at [line 7](https://github.com/hashicorp/vagrant/pull/9841/commits/4471bc48fb73c7d4774d7006a979dc97701147c3#diff-8cc7b2b0d78dd2501610391c086a8516R7) and fixed it before pushing, without thinking it was mentioned twice in the same document :laughing: